include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -6724,11 +6724,8 @@ namespace VectorTools
                const InVector        &fe_function,
                const Point<spacedim>      &point)
   {
-#ifdef DEBUG
-    const FiniteElement<dim> &fe = dof.get_fe();
-    Assert(fe.n_components() == 1,
+    Assert(dof.get_fe().n_components() == 1,
            ExcMessage ("Finite element is not scalar as is necessary for this function"));
-#endif
```

Comment:
(Which by no means implies that you are not absolutely right with any of these changes.)